Add default implementations to `RetryListener`

hpoettker opened the pull request:
With Spring Retry 2.0.0, the baseline was lifted from Java 6 to Java 17, which allows to now use default implementations in interfaces.

The only purpose of `RetryListenerSupport` is to provide default implementations for `RetryListener`. As such support implementations are no longer required, this PR deprecates `RetryListenerSupport` and adds its default implementations to `RetryListener`.

artembilan commented on src/test/java/org/springframework/retry/listener/RetryListenerSupportTests.java:
```diff
@@ -21,6 +21,7 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatNoException;
 
+@SuppressWarnings("deprecation")
```
I think it is better to rework tests in this class to deal with new code base.
I believe that this `RetryListenerSupport` will be removed in the next `2.1` version.
But that doesn't mean we will remove this test class.
So, let's see if just `new RetryListener()` is enough for unit tests in this class!

hpoettker replied:
There's already a class `RetryListenerTests`. I added the corresponding tests over there, and left this class as is and ready to be removed.

artembilan commented on src/main/java/org/springframework/retry/listener/RetryListenerSupport.java:
```diff
@@ -24,8 +24,9 @@
- *
+ * @deprecated in favor of the default implementations in {@link RetryListener}
  */
+@Deprecated
```
`since="2.0.1", forRemoval=true` ?

hpoettker replied:
done

artembilan reviewed (changes requested):
Please, add your name to the `@author` list of all the affected classes.
Also update their copyright to the current `2022`.

There is also a mentioning of `RetryListenerSupport` in the README.
Consider, please, revise that docs respectively.
(Not sure if we should mention its deprecation: just remove will be enough)

hpoettker commented:
@artembilan Thanks for the feedback!

The year in the headers is already 2022 in all affected classes. But I've added the `@author` tags. The line from the README is also removed.

artembilan commented:
@hpoettker ,

thank you for contribution; looking forward for more!
